### PR TITLE
Simplify template generation

### PR DESCRIPTION
**Pull request by salzhrani**
Although the current implementation relies on Ember views and templates which can be preferable but the fact of the matter getting strings out of templates and views is via a public API and has changed wildly between handlebars and htmlbars. Also it is more expensive to build a view and then getting its element html rather compared to constructing simple html strings and handing them to selectize. What this PR does is bind the render functions to functions passed as attributes.

If you think this is a valid approach I can update tweak it more, if not feel free to close it.

### CODE REVIEW

**Comment by miguelcobain** [comment]
I really like this!

Can you please change this PR to **not** remove the previous template/view logic?
Maybe there are folks who are still using it. :)

I would say the function takes precedence over the rest.
Also, a README update would be really nice as well.

**Comment by salzhrani** [comment]
Done.

File: addon/components/ember-selectize.js [hunk 1]
```diff
@@ -79,33 +79,39 @@ export default Ember.Component.extend({
...
       // infer the view name by camelizing selectize's function and appending a view suffix (overridable)
-      var viewPropertyName = camelize(item)+viewSuffix;
-      var viewToRender = get(this,viewPropertyName);
-
-      var self = this;
-      if(viewToRender){
+      var functionPropertyName = camelize(item) + 'Function';
+      var renderFunction = get(this,functionPropertyName);
+      if(renderFunction){
         // we have a view to render. set the function.
```
Comment (miguelcobain) [inline addon/components/ember-selectize.js]: Comments seem to be out of place?

**Comment by salzhrani** [comment]
ok, sorry about that.

**Comment by topaxi** [comment]
Nice, this is really useful! :+1: 
I was looking into monkey-patching the Selectize#render() method to make it ember compatible, but as long as Selectize doesn't support native DOM I can live with a solution like this PR. Probably even better than patching 3rdparty stuff :)

**Comment by miguelcobain** [comment]
@salzhrani Thanks!

**Comment by salzhrani** [comment]
Woot!!! Publish please.

**Comment by miguelcobain** [comment]
Done!

**Pull request merged**